static/app/components/copyToClipboardButton.tsx

```diff
@@ -0,0 +1,64 @@
+import {ComponentProps, useState} from 'react';
+
+import {Button, ButtonProps} from 'sentry/components/button';
+import Clipboard from 'sentry/components/clipboard';
+import {IconCopy} from 'sentry/icons';
+import {t} from 'sentry/locale';
+
+type Overwrite<T, U> = Pick<T, Exclude<keyof T, keyof U>> & U;
+
+type Props = {
+  text: string;
+  iconSize?: ComponentProps<typeof IconCopy>['size'];
+} & Overwrite<
+  ButtonProps,
+  Partial<
+    Pick<ButtonProps, 'aria-label'> & {onCopy: undefined | ((copiedText: string) => void)}
+  >
+>;
+
+export function CopyToClipboardButton({
+  iconSize = 'xs',
+  onCopy,
+  onMouseLeave,
+  text,
+  ...props
+}: Props) {
+  const [tooltipState, setTooltipState] = useState<'copy' | 'copied' | 'error'>('copy');
+
+  const tooltipTitle =
+    tooltipState === 'copy'
+      ? t('Copy')
+      : tooltipState === 'copied'
+      ? t('Copied')
+      : t('Unable to copy');
+
+  return (
+    <Clipboard
+      hideUnsupported
+      onSuccess={() => {
+        setTooltipState('copied');
+        onCopy?.(text);
+      }}
+      onError={() => {
+        setTooltipState('error');
+      }}
+      value={text}
+    >
+      <Button
+        aria-label={t('Copy')}
+        size={props.size || 'xs'}
```

Comment:
You're already providing a default
```suggestion
        size={props.size}
```

Reply:
oh yeah, i meant to remove the args default.

Reply:
wait args default seems OK, I would just remove this no?

Reply:
ah damnit, the button sizes are different from icon sizes. i'll just leave them both without defaults.
props: gotta set em all